File: example/pages/LoginPage.ts
import { BasePage } from 'conductor';
import type { Locator } from 'playwright';

export class LoginPage extends BasePage {
  private readonly usernameInput: Locator;
  private readonly passwordInput: Locator;
  private readonly submitButton: Locator;

  constructor(...args: ConstructorParameters<typeof BasePage>) {
    super(...args);
    this.usernameInput = this.page.locator('[data-testid="username"]');
    this.passwordInput = this.page.locator('[data-testid="password"]');
    this.submitButton = this.page.locator('[data-testid="login-submit"]');
  }

  async fillUsername(username: string): Promise<void> {
    await this.usernameInput.fill(username);
  }

  async fillPassword(password: string): Promise<void> {
    await this.passwordInput.fill(password);
  }

  async submit(): Promise<void> {
    await this.submitButton.click();
    await this.waitForLoad();
  }

  async login(username: string, password: string): Promise<void> {
    await this.navigate('/login');
    await this.fillUsername(username);
    await this.fillPassword(password);
    await this.submit();
  }
}

File: java/conductor-example/src/test/java/com/nouhouari/conductor/example/pages/LoginPage.java
package com.nouhouari.conductor.example.pages;

import com.microsoft.playwright.Page;
import com.nouhouari.conductor.config.EnvironmentConfig;
import com.nouhouari.conductor.pages.BasePage;

/** Java port of example/pages/LoginPage.ts. */
public class LoginPage extends BasePage {

    private final com.microsoft.playwright.Locator usernameInput;
    private final com.microsoft.playwright.Locator passwordInput;
    private final com.microsoft.playwright.Locator submitButton;

    public LoginPage(Page page, EnvironmentConfig config) {
        super(page, config);
        this.usernameInput = page.locator("[data-testid=\"username\"]");
        this.passwordInput = page.locator("[data-testid=\"password\"]");
        this.submitButton = page.locator("[data-testid=\"login-submit\"]");
    }

    public void fillUsername(String username) {
        usernameInput.fill(username);
    }

    public void fillPassword(String password) {
        passwordInput.fill(password);
    }

    public void submit() {
        submitButton.click();
        waitForLoad();
    }

    public void login(String username, String password) {
        navigate("/login");
        fillUsername(username);
        fillPassword(password);
        submit();
    }
}
